apps/block_scout_web/assets/js/omni.js

```diff
@@ -1,199 +1,318 @@
-  <div style="margin-top: 0.1rem;display: inline-block;font-weight: bolder;color: white;font-size: 1.2rem;padding-left: 0.5rem;">X-Rollup Transaction Details</div></div>
-  <pre style="font-size: 0.9rem;display: block;margin-top: 0.5rem;font-weight: bolder;color: white;padding-left: 1rem;">
-  ${JSON.stringify(details, null, 4)}
-  </pre>`;
-  return div;
-};
+    <div class="xchain-data__header">
+      <img class="xchain-data__header-logo" src="/images/omni_logo.svg" />
+      <div class="xchain-data__header-label">X-Rollup Transaction Details</div>
+      <div class="xchain-data__header-txcount">
+        ${
+          details.length > 1
+            ? details.length + ' transactions'
+            : '1 transaction'
+        }
+      </div>
+    </div>
+  `
+
+  for (const d of details) {
+    div.appendChild(xchainDetails(withExternalChainData(d)))
+  }
+
+  return div
+}
 
 // expected return {txHash1:true, txHas2:true...}
-const getXChainTxs = async (elements) => {
+const getXChainTxs = async elements => {
   try {
-    const response = await fetch(indexerUrl + "/checkOmniTxHashes", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        txHashes: elements,
-      }),
-    });
-    if (response.status == 400) {
-      await response.json().then(console.error);
-      return undefined;
+    const response = await fetch(indexerUrl + '/checkOmniTxHashes', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ txHashes: elements }),
+    })
+    if (response.status === 400) {
+      await response.json().then(console.error)
+      return undefined
     }
-    return response.json();
+    return response.json()
   } catch (err) {
-    console.error("error getting xchain txs: ", err);
-    return undefined;
+    console.error('error getting xchain txs: ', err)
+    return undefined
   }
-};
+}
 
-const getTxDetails = async (omniTxHash) => {
+const getTxDetails = async omniTxHash => {
   try {
-    const response = await fetch(indexerUrl + `/omniTx/${omniTxHash}`);
-    const data = await response.json();
+    const response = await fetch(indexerUrl + `/omniTx/${omniTxHash}`)
+    const data = await response.json()
     if (data.length === 0) {
-      return undefined;
+      return undefined
     }
-    return data;
+    return data
   } catch (err) {
-    console.error("error getting xchain tx details: ", omniTxHash, " ", err);
-    return undefined;
+    console.error('error getting xchain tx details: ', omniTxHash, ' ', err)
+    return undefined
   }
-};
+}
 
 const getAllTxHashElements = () => {
-  return document.querySelectorAll('[data-test="transaction_hash_link"]');
-};
+  return document.querySelectorAll('[data-test="transaction_hash_link"]')
+}
+
+const addXChainMarket = e => {
+  const marker = xchainMarker()
+  e.appendChild(marker)
+
+  // im sorry
```

Comment:
lol :D i assume there is no other selector to get element to set background color?

Reply:
there might be we can. let's mark that as a TODO